src/NServiceBus.RabbitMQ/Receiving/MessageConverter.cs

```diff
@@ -33,36 +33,44 @@ public string RetrieveMessageId(BasicDeliverEventArgs message, Dictionary<string
         public Dictionary<string, string> RetrieveHeaders(BasicDeliverEventArgs message)
         {
             var properties = message.BasicProperties;
+            var headers = properties.Headers;
```

Comment:
Unless you have great appetite to change, I'd leave as-is. But, you never know, I might send a refactoring PR a month from now 😉